### PR TITLE
v0.9.0 Feature (#375)

### LINKED ISSUE #375: Command: /armory
Option: name (required string)

Search for the weapon with the given name and return stats on the found weapon.

### PR DESCRIPTION
**Pull request by ntseng**
- /armory command

### CODE REVIEW

File: Source/Buttons/getgoldonfire.js [hunk 1]
```diff
@@ -1,24 +1,37 @@
...
+		adventure.gainGold(goldCount);
+    if (generateRandomNumber(adventure, 101, "general") > damageChance) { // deal damage on chance 50%
+      dealDamage(delver, null, 100, true, "Untyped", adventure).then(damageText => {
+        updateRoomHeader(adventure, interaction.message);
+        if (adventure.lives < 1) {
+          interaction.reply({ embeds: [completeAdventure(adventure, interaction.channel, { isSuccess: false, description: null })] });
+
+        } else {
+          interaction.update({ components: editButtons(interaction.message.components, { "getgoldonfire": { preventUse: true, label: `+${goldCount} gold`, emoji: "✔️" } }) })
+          interaction.channel.send(`${interaction.user} reaches in to grab some coin. Oh no! The gold burst into flames ${damageText}`);
```
Comment (doubledh) [inline Source/Buttons/getgoldonfire.js]: Does a delver still get the coins when the gold bursts into flames? The way the text reads makes me think the actual coins vaporize

File: Source/Commands/armory.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const Command = require('../../Classes/Command.js');
+const { getColor } = require('../elementHelpers.js');
+
+const options = [
+	{ type: "String", name: "weapon-name", description: "The name of the weapon (case-sensitive)", required: true, choices: {} }
```
Comment (doubledh) [inline Source/Commands/armory.js:5]: I feel that the case-sensitivity and non-fuzzy searching capabilities will mean that a player would be a lot more likely to find weapon details on the gitwiki, than by using an exact-string-matching command....  In what situation would a player use this command? From memory of another run, or from the current run, looking up the current weapons on hand (which should already be viewable), or perhaps looking up a weapon a friend picked up (where spelling/casing would be readily available)?

**Review by doubledh — COMMENTED** [review]
Couple of questions

File: Source/Commands/armory.js (hunk 2 above)
Comment (ntseng) [reply Source/Commands/armory.js:5]: I think fuzzy search would be a good feature. Not sure how to implement though.

File: Source/Buttons/getgoldonfire.js (hunk 1 above)
Comment (ntseng) [reply Source/Buttons/getgoldonfire.js]: Game balance-wise I think it'd be too punishing to have the gold evaporate. Maybe we should reword the feedback.

File: Source/Commands/armory.js (hunk 2 above)
Comment (ntseng) [reply Source/Commands/armory.js:5]: The other usecase I was thinking about was a player looking up what potential upgrades do. Anyways, given that we're planning a similarly structured command for enemies (and possibly events), fuzzy search seems like a good feature to invest in. It does look somewhat involved though: http://norvig.com/spell-correct.html

**Comment by ntseng** [comment]
Is blocking issue #243

File: Source/Commands/armory.js (hunk 2 above)
Comment (doubledh) [reply Source/Commands/armory.js:5]: oo, that fuzzy looks pretty practical, actually. Wonder if there's enough npm stuff to make that work in Node nearly verbatim. We can get to that later, though

**Review by doubledh — APPROVED** [review]

**Pull request merged**